corehq/apps/data_dictionary/util.py

```diff
@@ -155,6 +156,19 @@ def save_case_property(name, case_type, domain=None, data_type=None,
         update_fhir_resource_property(prop, fhir_resource_type, fhir_resource_prop_path, remove_path)
     prop.save()
 
+    # If caller has supplied non-None value for allowed_values, then
+    # synchronize the supplied dict (key=allowed_value, value=description)
+    # with the database stored values for this property.
+    if allowed_values is not None:
+        obj_pks = []
+        for allowed_value, av_desc in allowed_values.items():
+            av_obj, _ = CasePropertyAllowedValue.objects.update_or_create(
+                case_property=prop, allowed_value=allowed_value, defaults={"description": av_desc})
```

Comment:
HQ should do some kind of validation on the allowed values. How about django's `validate_slug`? It's possible there will be projects that want more flexibility, but that seems like a decent starting point.